core/src/main/java/com/airbnb/aerosolve/core/util/StringDictionary.java

```diff
@@ -0,0 +1,76 @@
+package com.airbnb.aerosolve.core.util;
+
+import com.airbnb.aerosolve.core.FeatureVector;
+import com.airbnb.aerosolve.core.DictionaryEntry;
+import com.airbnb.aerosolve.core.DictionaryRecord;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.io.Serializable;
+import java.util.*;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.Map.Entry;
+
+/**
+ * A class that maps strings to indices.
+ */
```

Comment:
perhaps add more details about this class, since it actually deals with both string features and float features